### ISSUE TITLE
Visualize with NetAnim

### ISSUE THREAD

**Issue by kaito2** [dragos-niculescu/ns3-wifi-infra-handoff#1]
Hi

I Ran wifi-infra-handoff.cc according to READMEfile

I want to visualize this simulation with NetAnim.

I added some code like below.

```
 //add animation config
  AnimationInterface::SetConstantPosition (serverNode, 100, 150); 
  AnimationInterface::SetConstantPosition (switchNode, 100, 100); 

  AnimationInterface anim ("a_mywifi1_animation.xml"); // Mandatory

  for (uint32_t i = 0; i < staNodes.size(); ++i)
    {
      for (uint32_t m = 0; m < staNodes[i].GetN(); ++m)
      {
        anim.UpdateNodeDescription (staNodes[i].Get(m), "STA"); // Optional
        anim.UpdateNodeColor (staNodes[i].Get(m), 255, 0, 0); // Optional
      }
    }
  for (uint32_t i = 0; i < apNodes.GetN (); ++i)
    {
      anim.UpdateNodeDescription (apNodes.Get (i), "AP"); // Optional
      anim.UpdateNodeColor (apNodes.Get (i), 0, 255, 0); // Optional
    }
  anim.UpdateNodeColor (serverNode, 255, 255, 0); // Optional
  anim.UpdateNodeColor (switchNode, 0, 0, 0); // Optional
  anim.EnablePacketMetadata (); // Optional
```
as a result , staNodes and these mobility are visualized, but packets between apNodes and staNodes aren't visualized.

How can i visualize these packet?

Thanks.

**Comment by dragos-niculescu** [comment]
Sorry, I haven't used xml animations, so I cannot be of any help :-(  

-- 
Dragoș

**Comment by apanand14** [comment]
Hello,
         I am trying to run your code in latest NS-3.29 version . Mobile trvales from 0 to 600m with 10m/s velocity. but it didn't proceed handover from AP0 to another any of AP1 or AP2. Do you recommend any changes for latest version or should I add few code line? 
Thank you in advance for your help.

**Comment by dragos-niculescu** [comment]
Hello,

I haven't try 3.29, but if you did, look into the pcap files and see if the association proceeds, or at which stage the process doesn't proceed as normal. 
I am willing to have a look at the pcap as well to help debug the issue. 

-- 
Dragoș

**Comment by apanand14** [comment]
Hello,
         Thank you for replying. I have attached Pcap file here. Can you please look into it? I haven't changed any Src code or i haven't used patch which you created. Thank you for help in advance.
[wifi-5-1.pcap.gz](https://github.com/dragos-niculescu/ns3-wifi-infra-handoff/files/3579522/wifi-5-1.pcap.gz)

[wifi-5-1.pcap.gz](https://github.com/dragos-niculescu/ns3-wifi-infra-handoff/files/3579520/wifi-5-1.pcap.gz)

**Comment by dragos-niculescu** [comment]
your pcap dopesn't have any packets. 
If you havent apply the patch, how can there be any handoff, unless they implemented it in newer versions.  

-- 
Dragoș

**Comment by apanand14** [comment]
Yes, When I run your code, then STA associates with AP0 only and it didn't handover to any AP1 and AP2 as you mentioned. I never used patch before. If you help me how to use it or how to add extra lines  of code for ns-3.29 version then it would be beneficial to me. your help will be appreciated. Thank you

**Comment by wenjun-y** [comment]
Hi, I have the same concern about how to use the patch file in my version of ns-3 (e.g., ns-3-dev) code, when I copy the patch file to the ns-3.xx folder and execute the command "patch -p1 < wifi-infra-handoff.patch" as you mentioned, the following errors came out:

> can't find file to patch at input line 4
> Perhaps you used the wrong -p or --strip option?
> The text leading up to this was:
> --------------------------
> |diff -Nur ns-3.22.orig/src/wifi/model/mac-low.cc ns-3.22/src/wifi/model/mac-low.cc
> |--- ns-3.22.orig/src/wifi/model/mac-low.cc	2015-02-06 01:46:22.000000000 +0200
> |+++ ns-3.22/src/wifi/model/mac-low.cc	2015-09-02 17:57:35.000000000 +0300
> --------------------------
> File to patch:

could you please elaborate further how should we use it properly? much appreciated.
